packages/vm/src/imports.rs

```diff
@@ -373,7 +380,7 @@ fn do_ed25519_verify<A: BackendApi, S: Storage, Q: Querier>(
     signature_ptr: u32,
     pubkey_ptr: u32,
 ) -> VmResult<u32> {
-    let message = read_region(&env.memory(), message_ptr, MESSAGE_MAX_LEN)?;
+    let message = read_region(&env.memory(), message_ptr, MAX_LENGTH_ED25519_MESSAGE)?;
```

Comment:
Note: because of this read limit it was impossible to hit cosmwasm-crypto's message length limit. So code 2 could never happen.